### DISCUSSION TITLE
Buderus BC400: HK1 Summer Mode not available

### DISCUSSION THREAD

**Discussion by Hank2419** [emsesp/EMS-ESP32#1971] [Q&A]
### PROBLEM DESCRIPTION

The entity “HK1 Summer Mode” ist visible in Home Assistant but is constantly shown as “unavailable”

### REQUESTED INFORMATION

- [x] Provide the output of http://ems-esp.local/api/system :

```lua
  System information output here:
{
    "System Info": {
        "version": "3.7.0-dev.23",
        "uptime": "018+14:38:58.443",
        "uptime (seconds)": 1607938,
        "platform": "ESP32-S3",
        "arduino": "ESP32 Arduino v2.0.16",
        "sdk": "v4.4.7-dirty",
        "free mem": 199,
        "max alloc": 179,
        "free caps": 8328,
        "used app": 2126,
        "free app": 3570,
        "partition": "app1",
        "reset reason": "Software reset CPU / Software reset CPU"
    },
    "Network Info": {
        "network": "WiFi",
        "hostname": "ems-esp",
        "RSSI": -60,
        "BSSID": "set",
        "TxPower setting": 78,
        "static ip config": false,
        "low bandwidth": false,
        "disable sleep": false,
        "enable MDNS": true,
        "enable CORS": false,
        "AP provision mode": "disconnected",
        "AP security": "wpa2",
        "AP ssid": "ems-esp"
    },
    "NTP Info": {
        "NTP status": "connected",
        "enabled": true,
        "server": "time.google.com",
        "tz label": "Europe/Berlin"
    },
    "MQTT Info": {
        "MQTT status": "connected",
        "MQTT publishes": 348624,
        "MQTT queued": 0,
        "MQTT publish fails": 0,
        "MQTT connects": 1,
        "enabled": true,
        "client id": "ems-esp2",
        "keep alive": 60,
        "clean session": true,
        "entity format": 3,
        "base": "ems-esp2",
        "discovery prefix": "homeassistant",
        "discovery type": 0,
        "nested format": 1,
        "ha enabled": true,
        "mqtt qos": 0,
        "mqtt retain": false,
        "publish time heartbeat": 60,
        "publish time boiler": 10,
        "publish time thermostat": 10,
        "publish time solar": 10,
        "publish time mixer": 10,
        "publish time water": 10,
        "publish time other": 10,
        "publish time sensor": 10,
        "publish single": false,
        "publish2command": false,
        "send response": false
    },
    "Syslog Info": {
        "enabled": false
    },
    "Sensor Info": {
        "temperature sensors": 0,
        "temperature sensor reads": 0,
        "temperature sensor fails": 0,
        "analog sensors": 0,
        "analog sensor reads": 0,
        "analog sensor fails": 0
    },
    "API Info": {
        "API calls": 0,
        "API fails": 0
    },
    "Bus Info": {
        "bus status": "connected",
        "bus protocol": "Buderus",
        "bus telegrams received (rx)": 5054715,
        "bus reads (tx)": 1152358,
        "bus writes (tx)": 3,
        "bus incomplete telegrams": 148,
        "bus reads failed": 0,
        "bus writes failed": 0,
        "bus rx line quality": 100,
        "bus tx line quality": 100
    },
    "Settings": {
        "board profile": "S32S3",
        "locale": "de",
        "tx mode": 2,
        "ems bus id": 11,
        "shower timer": false,
        "shower min duration": 180,
        "shower alert": false,
        "hide led": false,
        "notoken api": false,
        "readonly mode": false,
        "fahrenheit": false,
        "dallas parasite": false,
        "bool format": 1,
        "bool dashboard": 1,
        "enum format": 1,
        "analog enabled": true,
        "telnet enabled": true,
        "max web log buffer": 50,
        "web log buffer": 50
    },
    "Devices": [
        {
            "type": "boiler",
            "name": "WSW196i",
            "device id": "0x08",
            "product id": 172,
            "version": "03.02",
            "entities": 137,
            "handlers received": "0xBF 0xC2 0xD1 0xE3 0xE4 0xE5 0xE9 0x0494 0x0495 0x048F",
            "handlers fetched": "0x14 0xE6 0xEA 0x048D 0x048A 0x04A2 0x0485 0x0486 0x0492 0x0488 0x0484 0x048B 0x0491 0x0499 0x049C 0x049D 0x02CC 0x04A5",
            "handlers pending": "0x10 0x11 0x15 0x1C 0x18 0x19 0x1A 0x35 0x16 0x33 0x34 0x26 0x2A 0x28 0x04AE 0x04AF",
            "handlers ignored": "0xF7 0x04C0 0x02EA 0xE7 0x04F2 0x02D6 0x0489 0x04AC 0x0240 0x0291 0x029B 0x0497 0x02E0 0x049F 0x04A0 0x2E 0x3B 0x17 0x36 0xF6"
        },
        {
            "type": "thermostat",
            "name": "WSW196i/BC400",
            "device id": "0x10",
            "product id": 253,
            "version": "47.03",
            "entities": 66,
            "handlers received": "0x06 0x02BA 0x02BB 0x02BC 0x031D 0x0267",
            "handlers fetched": "0x02A5 0x02B9 0x02AF 0x029B 0x0471 0x0467 0x0291 0x0292 0x0293 0x0294 0x02F5 0x023A 0x0240 0xBB 0x023E",
            "handlers pending": "0xA3 0xA2 0x12 0x13 0x02A6 0x02B0 0x029C 0x0472 0x02A7 0x02B1 0x029D 0x0473 0x02A8 0x02B2 0x029E 0x0474 0x02A9 0x02BD 0x02B3 0x029F 0x0475 0x02AA 0x02BE 0x02B4 0x02A0 0x0476 0x02AB 0x02BF 0x02B5 0x02A1 0x0477 0x02AC 0x02C0 0x02B6 0x02A2 0x0478 0x02CC 0x02CE 0x0468 0x02D0 0x0469 0x02D2 0x046A 0x02F6 0x031B 0x031E 0x0269",
            "handlers ignored": "0x0507 0x0508 0x0509 0x050A 0x059D 0xBF 0xF7 0xF6"
        },
        {
            "type": "thermostat",
            "name": "RC100H",
            "device id": "0x38",
            "product id": 200,
            "version": "40.07",
            "entities": 3,
            "handlers received": "0x042B 0x047B",
            "handlers fetched": "0x0273",
            "handlers pending": "0x0A6A",
            "handlers ignored": "0xBF 0x04A2"
        },
        {
            "type": "gateway",
            "name": "WiFi module",
            "device id": "0x48",
            "product id": 252,
            "version": "08.01",
            "entities": 0,
            "handlers ignored": "0x2040 0xF9 0x0495 0x0494 0x3B 0x0CBD"
        }
    ]
}

```
Will be provided later if needed

### TO REPRODUCE

The error is constantly visible if entity is displayed in HA

### EXPECTED BEHAVIOUR

The entity should be available with a valid status

### SCREENSHOTS
<img width="355" alt="image" src="https://github.com/user-attachments/assets/e0cf24c1-968a-48c5-9ff3-10cbfca45cd3">
<img width="603" alt="image" src="https://github.com/user-attachments/assets/08aa6e9d-5968-4e1d-82d4-4e2919d1eb25">

### ADDITIONAL CONTEXT

None

**Comment by proddy** [comment]
please attach the support info

**Comment by proddy** [comment]
and see what value is shown in the EMS-ESP - either on the WebUI or via the Console with `show values` or an API call like `/api/thermostat/info`

**Comment by Hank2419** [comment]
> please attach the support info

I have added this info above



> and see what value is shown in the EMS-ESP - either on the WebUI or via the Console with `show values` or an API call like `/api/thermostat/info`

{
    "name": "summermode",
    "fullname": "HK1 Sommerbetrieb",
    "circuit": "hc1",
    "type": "enum",
    "enum": [
        "Winter",
        "Sommer"
    ],
    "readable": true,
    "writeable": false,
    "visible": true,
    "value": "not set"
}

![image](https://github.com/user-attachments/assets/d6d09dbb-2598-4f50-b481-00c6c11b4967)

**Comment by proddy** [comment]
There is no value (it says not set) so its not sent to HA.  Modes for BC400 was recently added. i would upgrade to the latest dev, try again and if it still doesn't work then help us find the right value in the telegram - https://emsesp.github.io/docs/Troubleshooting/#im-missing-certain-data-from-an-ems-device

**Comment by Hank2419** [comment]
> There is no value (it says not set) so its not sent to HA. Modes for BC400 was recently added. i would upgrade to the latest dev, try again and if it still doesn't work then help us find the right value in the telegram - https://emsesp.github.io/docs/Troubleshooting/#im-missing-certain-data-from-an-ems-device

I’m on esp firmware version 3.7.0-dev.23 (ESP32-S3)

Since I have installed ESP the summer mode in the BC400 has never changed and remains on “Sommer”. Next change to “Winter” will be next autumn when temperature is below 15°. Could that be the reason that the value is not transmitted?

I have updated now to the latest build and restarted. Interestingly during the restart process the summer mode is displayed correctly as “Sommer” for a short time before it turnes back to unavailable. 

 can reproduce this behavior for every restart:
 
![image](https://github.com/user-attachments/assets/8f229c2e-39e8-4e8d-b2a0-8878eeff1375)

this is how it looks like some minutes after restart:

![image](https://github.com/user-attachments/assets/522cf6a7-9cb4-427a-8a40-2a0be2bf0c47)


![image](https://github.com/user-attachments/assets/265a7f81-d760-4def-abc7-dd1f7bc95d2e)

**Comment by Hank2419** [comment]
Another observation: during restart the different entities are changing their status. First Sommer mode is valid and other entities are unavailable. At last when “Arbeitsweise” turns available at the very same moment “summer mode” turns unavailable. Is this coincidence?

![image](https://github.com/user-attachments/assets/5830b64c-a3a9-4b9a-9387-d503e4410712)

**Comment by MichaelDvP** [comment]
See #550 and #566, heatpumps does not show summermode, they show operatingmode instead. Since more than 2 years now.

**Comment by Hank2419** [comment]
> See #550 and #566, heatpumps does not show summermode, they show operatingmode instead. Since more than 2 years now.

In my case there is a different behavior of the signal. It is the only signal from the Buderus heat pump that is unavailable. All other approx 400 signals are available (thereof many with operating modes) and seem to work as the should. Also I found out, that during restart process of the ESP Gateway for a couple of minutes the values is correctly displayed in HA (see screenshot above) before it turns back to “unavailable”. However the signal is for me only “nice to have” and I fully depend on experts to solve the issue. Thanks to all who provide support!

**Comment by Hank2419** [comment]
I wonder if other users experience the same problem with Buderus Heatpump. I have updated to the latest firmware 3.7.2 but issue still persist.
